src/Razor/test/RazorLanguage.Test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -3418,6 +3418,67 @@ @using Test.Shared
             CompileToAssembly(generated);
         }
 
+        [Fact]
+        public void GenericComponent_NonGenericEventCallback_TypeInference()
+        {
+            // Arrange
+            AdditionalSyntaxTrees.Add(Parse(@"
+using Microsoft.AspNetCore.Components;
+
+namespace Test
+{
+    public class MyEventArgs { }
+
+    public class MyComponent<TItem> : ComponentBase
+    {
+        [Parameter] public TItem Item { get; set; }
+        [Parameter] public EventCallback MyEvent { get; set; }
+    }
+}
+"));
+            // Act
+            var generated = CompileToCSharp(@"
+@using Test
+<MyComponent Item=""3"" MyEvent=""x => {}"" />
+");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+        }
+
+        [Fact]
+        public void GenericComponent_GenericEventCallback_TypeInference()
+        {
+            // Arrange
+            AdditionalSyntaxTrees.Add(Parse(@"
+using Microsoft.AspNetCore.Components;
+
+namespace Test
+{
+    public class MyEventArgs { }
+
+    public class MyComponent<TItem> : ComponentBase
+    {
+        [Parameter] public TItem Item { get; set; }
+        [Parameter] public EventCallback<MyEventArgs> MyEvent { get; set; }
```

Comment:
Need a similar test with `EventCallback<TItem>`.

Reply:
Added one below. However you have to use a delegate and not a method group due to type inference. Makes it ever so slightly less interesting.